### ISSUE TITLE
ethminer (0.13.0) fails but continues to log as though it had not

### ISSUE THREAD

**Issue by michael-pesce** [ethereum-mining/ethminer#707]
I detect crashes by watching the log file. When output is no longer written fox X seconds I assume a crash, kill ethminer (if it's still running) and start it again. I just updated to 0.13.0 and this morning noticed that one of my miners went down last night but never restarted. When I went to the log I saw it was still being written. However: 

1) The hash rate was not changing at all and was weirdly low (by ~2MH/s per card), and
2) In nvidia-smi it was clear that the cards were doing nothing as they were drawing low power

So, it appears there was a crash but ethminer just chugged along as though nothing happened though it was effectively crashed. 

LOG 
```  m  09:06:37|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:25
  m  09:06:38|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:25
  m  09:06:39|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:25
  m  09:06:40|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:25
  ℹ  09:06:41|stratum   Received new job #c0764893  seed: #a8784097a4d03c2d2ac6a3a2beebd060  target: #0000000112e0be826d694b2e
  m  09:06:41|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:25
  m  09:06:42|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:43|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:44|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:45|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:46|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:47|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:48|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  ℹ  09:06:49|stratum   Received new job #f04de2d5  seed: #a8784097a4d03c2d2ac6a3a2beebd060  target: #0000000112e0be826d694b2e
  m  09:06:49|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:50|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:51|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:52|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:53|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:54|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
  m  09:06:55|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 46:26
```

NVIDIA-SMI
```
Tue Feb  6 09:06:58 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 384.111                Driver Version: 384.111                   |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 1070    On   | 00000000:01:00.0  On |                  N/A |
| 90%   34C    P2    34W / 115W |   2666MiB /  8110MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 1070    On   | 00000000:02:00.0 Off |                  N/A |
| 90%   33C    P2    36W / 115W |   2506MiB /  8114MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  GeForce GTX 1070    On   | 00000000:03:00.0 Off |                  N/A |
| 90%   30C    P2    34W / 115W |   2506MiB /  8114MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  GeForce GTX 1070    On   | 00000000:04:00.0 Off |                  N/A |
| 90%   32C    P2    33W / 115W |   2506MiB /  8114MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   4  GeForce GTX 1070    On   | 00000000:05:00.0 Off |                  N/A |
| 90%   32C    P2    34W / 115W |   2506MiB /  8114MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   5  GeForce GTX 1070    On   | 00000000:06:00.0 Off |                  N/A |
| 90%   29C    P2    34W / 115W |   2506MiB /  8114MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1028      G   /usr/lib/xorg/Xorg                           126MiB |
|    0      1665      G   compiz                                        41MiB |
|    0      9971      C   /home/michael/bin/ethminer                  2487MiB |
|    1      1028      G   /usr/lib/xorg/Xorg                             7MiB |
|    1      9971      C   /home/michael/bin/ethminer                  2487MiB |
|    2      1028      G   /usr/lib/xorg/Xorg                             7MiB |
|    2      9971      C   /home/michael/bin/ethminer                  2487MiB |
|    3      1028      G   /usr/lib/xorg/Xorg                             7MiB |
|    3      9971      C   /home/michael/bin/ethminer                  2487MiB |
|    4      1028      G   /usr/lib/xorg/Xorg                             7MiB |
|    4      9971      C   /home/michael/bin/ethminer                  2487MiB |
|    5      1028      G   /usr/lib/xorg/Xorg                             7MiB |
|    5      9971      C   /home/michael/bin/ethminer                  2487MiB |
+-----------------------------------------------------------------------------+```

**Comment by piotr-dobrogost** [comment]
Compare with issue #651

**Comment by michael-pesce** [comment]
Some more data. 

Went back in logs and found this: 

```
ℹ  02:25:36|stratum   Received new job #00163275  seed: #b683d2a971567602a1931b76b9fb447d  target: #0000000112e0be826d694b2e
  ℹ  02:25:37|cuda-2    Solution found; Submitted to us1.ethermine.org
  ℹ  02:25:37|cuda-2    Nonce: 0x1ccaf7090eb34ead
  ℹ  02:25:37|stratum    Accepted. 
  m  02:25:37|ethminer  Speed 186.02 Mh/s    gpu/0 30.81  gpu/1 30.72  gpu/2 31.19  gpu/3 31.10  gpu/4 31.10  gpu/5 31.10  [A6785+20:R1+0:F0] Time: 39:44
  ℹ  02:25:37|stratum   Received new job #aaebbc32  seed: #b683d2a971567602a1931b76b9fb447d  target: #0000000112e0be826d694b2e
  m  02:25:38|ethminer  Speed 185.22 Mh/s    gpu/0 30.75  gpu/1 30.54  gpu/2 30.96  gpu/3 30.96  gpu/4 30.96  gpu/5 31.06  [A6785+20:R1+0:F0] Time: 39:44
  ℹ  02:25:38|stratum   Received new job #945dfde8  seed: #b683d2a971567602a1931b76b9fb447d  target: #0000000112e0be826d694b2e
  m  02:25:39|ethminer  Speed 184.80 Mh/s    gpu/0 30.64  gpu/1 30.54  gpu/2 30.85  gpu/3 30.85  gpu/4 30.96  gpu/5 30.96  [A6785+20:R1+0:F0] Time: 39:44
  m  02:25:40|ethminer  Speed 184.96 Mh/s    gpu/0 30.58  gpu/1 30.58  gpu/2 30.87  gpu/3 30.97  gpu/4 30.97  gpu/5 30.97  [A6785+20:R1+0:F0] Time: 39:44
  m  02:25:41|ethminer  Speed 185.58 Mh/s    gpu/0 30.72  gpu/1 30.63  gpu/2 31.01  gpu/3 31.01  gpu/4 31.10  gpu/5 31.10  [A6785+20:R1+0:F0] Time: 39:44
  m  02:25:42|ethminer  Speed 185.48 Mh/s    gpu/0 30.82  gpu/1 30.63  gpu/2 31.01  gpu/3 31.01  gpu/4 31.01  gpu/5 31.01  [A6785+20:R1+0:F0] Time: 39:44
  m  02:25:43|ethminer  Speed 185.39 Mh/s    gpu/0 30.72  gpu/1 30.63  gpu/2 31.01  gpu/3 31.01  gpu/4 31.01  gpu/5 31.01  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:44|ethminer  Speed 185.48 Mh/s    gpu/0 30.72  gpu/1 30.63  gpu/2 31.01  gpu/3 31.10  gpu/4 31.01  gpu/5 31.01  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:45|ethminer  Speed 185.55 Mh/s    gpu/0 30.82  gpu/1 30.64  gpu/2 31.00  gpu/3 31.09  gpu/4 31.00  gpu/5 31.00  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:46|ethminer  Speed 186.28 Mh/s    gpu/0 30.87  gpu/1 30.78  gpu/2 31.13  gpu/3 31.22  gpu/4 31.13  gpu/5 31.13  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:47|ethminer  Speed 186.10 Mh/s    gpu/0 30.87  gpu/1 30.70  gpu/2 31.13  gpu/3 31.13  gpu/4 31.13  gpu/5 31.13  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:48|ethminer  Speed 186.19 Mh/s    gpu/0 30.87  gpu/1 30.78  gpu/2 31.13  gpu/3 31.13  gpu/4 31.13  gpu/5 31.13  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:49|ethminer  Speed 186.27 Mh/s    gpu/0 30.89  gpu/1 30.81  gpu/2 31.14  gpu/3 31.14  gpu/4 31.14  gpu/5 31.14  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:50|ethminer  Speed 186.78 Mh/s    gpu/0 30.99  gpu/1 30.83  gpu/2 31.24  gpu/3 31.24  gpu/4 31.24  gpu/5 31.24  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:51|ethminer  Speed 186.83 Mh/s    gpu/0 30.91  gpu/1 30.91  gpu/2 31.31  gpu/3 31.23  gpu/4 31.23  gpu/5 31.23  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:52|ethminer  Speed 187.37 Mh/s    gpu/0 31.03  gpu/1 30.95  gpu/2 31.40  gpu/3 31.33  gpu/4 31.33  gpu/5 31.33  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:53|ethminer  Speed 187.35 Mh/s    gpu/0 31.09  gpu/1 30.94  gpu/2 31.31  gpu/3 31.31  gpu/4 31.31  gpu/5 31.39  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:54|ethminer  Speed 187.75 Mh/s    gpu/0 31.13  gpu/1 30.99  gpu/2 31.41  gpu/3 31.41  gpu/4 31.41  gpu/5 31.41  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:55|ethminer  Speed 187.75 Mh/s    gpu/0 31.13  gpu/1 30.99  gpu/2 31.41  gpu/3 31.41  gpu/4 31.41  gpu/5 31.41  [A6785+20:R1+0:F0] Time: 39:45
CUDA error in func 'search' at line 489 : an illegal memory access was encountered.
  ✘  02:25:55|cuda-3    Error CUDA mining: an illegal memory access was encountered
CUDA error in func 'search' at line 489 : an illegal memory access was encountered.
  ✘  02:25:55|cuda-2    Error CUDA mining: an illegal memory access was encountered
CUDA error in func 'search' at line 489 : an illegal memory access was encountered.
  ✘  02:25:55|cuda-0    Error CUDA mining: an illegal memory access was encountered
CUDA error in func 'search' at line 489 : an illegal memory access was encountered.
  ✘  02:25:55|cuda-1    Error CUDA mining: an illegal memory access was encountered
CUDA error in func 'search' at line 489 : an illegal memory access was encountered.
CUDA error in func 'search' at line 489 : an illegal memory access was encountered.
  ✘  02:25:55|cuda-5    Error CUDA mining: an illegal memory access was encountered
  ✘  02:25:55|cuda-4    Error CUDA mining: an illegal memory access was encountered
  m  02:25:56|ethminer  Speed 181.02 Mh/s    gpu/0 30.01  gpu/1 29.94  gpu/2 30.29  gpu/3 30.22  gpu/4 30.29  gpu/5 30.29  [A6785+20:R1+0:F0] Time: 39:45
  ℹ  02:25:56|stratum   Received new job #f918ee7d  seed: #b683d2a971567602a1931b76b9fb447d  target: #0000000112e0be826d694b2e
  m  02:25:57|ethminer  Speed 180.13 Mh/s    gpu/0 29.86  gpu/1 29.78  gpu/2 30.10  gpu/3 30.02  gpu/4 30.18  gpu/5 30.18  [A6785+20:R1+0:F0] Time: 39:45
  m  02:25:58|ethminer  Speed 179.40 Mh/s    gpu/0 29.72  gpu/1 29.64  gpu/2 29.99  gpu/3 29.90  gpu/4 30.07  gpu/5 30.07  [A6785+20:R1+0:F0] Time: 39:45
  ℹ  02:25:58|stratum   Received new job #c0c9743c  seed: #b683d2a971567602a1931b76b9fb447d  target: #0000000112e0be826d694b2e
  m  02:25:59|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 39:45
  m  02:26:00|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 39:45
  m  02:26:01|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 39:45
  m  02:26:02|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 39:45
  m  02:26:03|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 39:45
  m  02:26:04|ethminer  Speed 177.72 Mh/s    gpu/0 29.48  gpu/1 29.37  gpu/2 29.69  gpu/3 29.58  gpu/4 29.79  gpu/5 29.79  [A6785+20:R1+0:F0] Time: 39:45
```

It appears this error was the crash:

```
CUDA error in func 'search' at line 489 : an illegal memory access was encountered.
```

But, ethminer keeps going which seems like a defect.

**Comment by traderbighouse** [comment]
I wish to confirm that I am also experiencing this problem on my side after upgrading to this version today. Same issue as posted by @michael-pesce above

**Comment by dwayneem** [comment]
Also experiencing this issue with 0.13.0

**Comment by piotr-dobrogost** [comment]
Duplicate of issue #651 which is a duplicate of issue #632

**Comment by DeadManWalkingTO** [comment]
After #757 (added --exit parameter to exit whenever an error occurred) you can use a watchdog.

Here is my [ETHminerWatchDogDmW](https://github.com/DeadManWalkingTO/ETHminerWatchDogDmW) Windows7/8/10 [32/64] & Linux (Any Dist/Any Ver/Any Arch) (#735).

Check and feedback please.
Thank you!

**Comment by AndreaLanfranchi** [comment]
No feed back.
